### ISSUE TITLE
registerDeletion not found error

### ISSUE THREAD

**Issue by spormeon** [prebid/Prebid.js#9425]
I'm testing out the GPP module, first it seems not to work with quantcast cmp, presume their cmp not support it yet? 

Than if I comment out the gpp refrences but leave in the gpp module
consentManagement: {
						gdpr: {
							cmpApi: 'iab',
							timeout: 25000,
							// allowAuctionWithoutConsent: false,
							defaultGdprScope: true,
                            // rules: [{        // these are the default values
                            //     purpose: "storage",
                            //     enforcePurpose: true,
                            //     enforceVendor: true
                            //   },{
                            //     purpose: "basicAds",
                            //     enforcePurpose: true,
                            //     enforceVendor: true
                            //   },{
                            //     purpose: "measurement",
                            //     enforcePurpose: true,
                            //     enforceVendor: true
                            //   }]
						},
						usp: {
							cmpApi: 'iab',
							timeout: 13500
						},
                        // gpp: {
                        //     cmpApi: 'iab',
                        //     timeout: 25000
                        // }
					},

I'm left with an error, which looks like its something to do with Amazon to me, but no real clue? ( the error was also there when the prebid warning, no gpp cmp found was being thrown, when the gpp consent setting we're in)

![DevTools_-_cdn_adysis_com_VideoDemo_videofloat_html_and_Video_Float_Demo](https://user-images.githubusercontent.com/8503167/212497603-df994d63-ffa2-4a82-93f4-2543d3e41f22.png)

**Comment by spormeon** [comment]
update:  this error seems to appear somehwre between V7.17.0 and V7.32.0 and its not related to gpp module by the looks, I took out gpp module in 7.32.0 and its still there.  Roll back to 7.17.0 and its not there, so not a scooby doo what it is or where come from

**Comment by spormeon** [comment]
i dont think it is now, as when i TAKE OUT the gpp module its still there in 7.32.0,  I'm gonna itterate up the V's to find where it apears, bare with me.

**Comment by dgirardi** [comment]
This seems to be related to the USP data deletion feature ('registerDeletion') but it's not clear how. Could you provide a test page?

**Comment by patmmccann** [comment]
It seems https://github.com/prebid/Prebid.js/blob/d646c55ced9f8ee6f596df1cd717e72b1c0ebcd7/modules/consentManagementUsp.js#L119 is likely making your cmp barf bc it doesnt handle register deletion. It still seems to be a q for quantcast

**Comment by spormeon** [comment]
should it actually even do this, im in uk and usp is nowhere near me, shouldnt this thing fire when "someone" is in the relevant area/ territory?

**Comment by dgirardi** [comment]
it should attempt it only if you have the `consentManagementUsp` module included _and_ there is a USP CMP on the page. Even when it does, it should not fail as you describe - something is off, that's why a test page would be helpful.

**Comment by dgirardi** [comment]
Thank you @idod-wordline. It looks like the CMP on that page does not recognize the `registerDeletion` and `performDeletion` USP API commands, posted #9434 to handle that.

I am a bit confused by the situation as the [spec](https://github.com/InteractiveAdvertisingBureau/USPrivacy/blob/master/CCPA/Data%20Deletion%20Request%20Handling.md) dates from 2020 and the CMP source does have _some_ references to data deletion; the website itself also references it, but I don't see any way to actually request the deletion.

**Comment by spormeon** [comment]
is this any use, you have to add an extra bit of code to the bottom of the page for quantcast to show the USP "do not sell my data" link etc: https://www.specktra.net/
![view-source_https___www_specktra_net](https://user-images.githubusercontent.com/8503167/213226035-f8417247-cb83-4f88-a4d2-91210e2e84ee.png)

**Comment by idod-wordline** [comment]
> is this any use, you have to add an extra bit of code to the bottom of the page for quantcast to show the USP "do not sell my data" link etc: https://www.specktra.net/ ![view-source_https___www_specktra_net](https://user-images.githubusercontent.com/8503167/213226035-f8417247-cb83-4f88-a4d2-91210e2e84ee.png)

Does this somehow address the error?

**Comment by idod-wordline** [comment]
@dgirardi Could it have something to do with this merged PR: https://github.com/prebid/Prebid.js/pull/9116 ?

It seems this error started showing up for us around that time.

**Comment by spormeon** [comment]
no, as its on my test page , where the error still was

**Comment by spormeon** [comment]
I'm still getting the error with V7.33.0

**Comment by dgirardi** [comment]
@spormeon the error should be inconsequential, at least based on the test page provided by @idod-wordline. If it isn't (something is not working, apart the console warning) feel free to reopen this issue, but please provide a test page and steps to reproduce.

**Comment by idod-wordline** [comment]
@dgirardi It was inconsequential even before your fix, in that sense, but this error show up A LOT for us and that in itself is very consequential for us as it sabotages our monitoring (had to disable sentry because it was flooding us and ramping up the cost) and gets in the way of debugging real issues. Can it be suppressed in any way?

**Comment by dgirardi** [comment]
We could introduce a configuration flag to disable the feature. My understanding however is that the CMP is out of spec and the feature is or will shortly be required for compliance in the US. @patmmccann ?

**Comment by spormeon** [comment]
as a chip in.  Quantcast are one of the big cmp's, 000's using them, have you guys got noway to colaborate with them on it?

**Comment by dgirardi** [comment]
I did reach out to Quantcast, but I encourage you to do the same;  if you are a customer you have a better channel to address support questions such as whether it's possible to suppress that error message.

**Comment by patmmccann** [comment]
@spormeon we have plenty of contacts at quantcast; don't worry we'll get ahold of them

**Comment by patmmccann** [comment]
If an example of uspapi cannot handle the registerDeletion event it is out of compliance with https://github.com/InteractiveAdvertisingBureau/USPrivacy/blob/master/CCPA/Data%20Deletion%20Request%20Handling.md#registerdeletion

This is pretty cut and dry an issue with quantcast software and i think the error is useful, informative, and should remain

**Comment by patmmccann** [comment]
Quantcast reports fix in progress

**Comment by idod-wordline** [comment]
> Quantcast reports their fix is expected to be live Feb 6

@patmmccann Error is still present, can you please find out if Quantcast's fix was published? Or get from them a url where this issue can be tracked?

**Comment by spormeon** [comment]
the function registerdeletion error seems to have gone, but its now throwing this error instead after quantcast V46 release:

![DevTools_-_cdn_adysis_com_VideoDemo_videofloat_html](https://user-images.githubusercontent.com/8503167/217646032-dbe53844-e49f-4efb-bcfe-af859b9f39cc.png)

@idod-wordline , you seing the same new error now?

**Comment by idod-wordline** [comment]
No, for us all seems to be working well at this time. Thanks for everyone's
help.

@spormeon Feel free to take a look at our config via Professor Prebid.

**Comment by spormeon** [comment]
you havent actually got the GPP element configed in your setting.  So i'm guessing they fixed the error but they haven't made the actual cmp GPP compatible yet, more months to wait i guess, if wanna use quantcast for GPP:

![prebidVideo4_29_0_js_—_Prebid_Publishers__Workspace_](https://user-images.githubusercontent.com/8503167/217651599-fc2cd454-d731-4ef7-9125-74f765755f91.png)
